lib/msf/core/handler/reverse_http.rb

```diff
@@ -260,9 +260,7 @@ def setup_handler
   def stop_handler
     if self.service
       self.service.remove_resource((luri + "/").gsub("//", "/"))
-      if self.service.resources.empty? && self.sessions == 0
-        Rex::ServiceManager.stop_service(self.service)
-      end
+      Rex::ServiceManager.stop_service(self.service)
```

Comment:
Similar to the below question, should this be something akin to:

```suggestion
      self.service.deref if self.service
      self.service = nil
```